github/resource_github_organisation_custom_properties.go

```diff
@@ -0,0 +1,141 @@
+				Description: "The name of the custom property",
+			},
+			"value_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The type of the custom property",
+			},
+			"required": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether the custom property is required",
+			},
+			"default_value": {
+				Type:        schema.TypeString,
+				Description: "The default value of the custom property",
+				Optional:    true,
+				Computed:    true,
+			},
+			"description": {
+				Description: "The description of the custom property",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"allowed_values": {
+				Description: "The allowed values of the custom property",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceGithubCustomPropertiesCreate(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.Background()
+	client := meta.(*Owner).v3client
+	ownerName := meta.(*Owner).name
+
+	propertyName := d.Get("property_name").(string)
+	valueType := d.Get("value_type").(string)
+	required := d.Get("required").(bool)
+	defaultValue := d.Get("default_value").(string)
+	description := d.Get("description").(string)
+	allowedValues := d.Get("allowed_values").([]interface{})
+	var allowedValuesString []string
+	for _, v := range allowedValues {
+		allowedValuesString = append(allowedValuesString, v.(string))
+	}
+
+	customProperty, _, err := client.Organizations.CreateOrUpdateCustomProperty(ctx, ownerName, d.Get("property_name").(string), &github.CustomProperty{
+		PropertyName:  &propertyName,
+		ValueType:     valueType,
+		Required:      &required,
+		DefaultValue:  &defaultValue,
+		Description:   &description,
+		AllowedValues: allowedValuesString,
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*customProperty.PropertyName)
+	return resourceGithubCustomPropertiesRead(d, meta)
+}
+
+func resourceGithubCustomPropertiesRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.Background()
+	client := meta.(*Owner).v3client
+	ownerName := meta.(*Owner).name
+
+	customProperty, _, err := client.Organizations.GetCustomProperty(ctx, ownerName, d.Get("property_name").(string))
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*customProperty.PropertyName)
+	d.Set("allowed_values", customProperty.AllowedValues)
+	d.Set("default_value", customProperty.DefaultValue)
+	d.Set("description", customProperty.Description)
+	d.Set("property_name", customProperty.PropertyName)
+	d.Set("required", customProperty.Required)
+	d.Set("value_type", customProperty.ValueType)
+
+	return nil
+}
+
+func resourceGithubCustomPropertiesUpdate(d *schema.ResourceData, meta interface{}) error {
+	if err := resourceGithubCustomPropertiesCreate(d, meta); err != nil {
+		return err
+	}
+	return resourceGithubTeamRead(d, meta)
```

Comment:
you probably want to return the resourceGithubCustomPropertiesRead(d, meta) value, I guess this is a typo